{
  List<Throwable> errors=Lists.newArrayList();
  try {
    next.evaluate();
  }
 catch (  Throwable e) {
    errors.add(e);
  }
 finally {
    try {
      after.evaluate();
    }
 catch (    Throwable e) {
      errors.add(e);
    }
  }
  MultipleFailureException.assertEmpty(errors);
}
